packages/valibot/src/__tests__/contract.test.ts

```diff
@@ -0,0 +1,121 @@
+import * as v from 'valibot';
+import { describe, test, expect } from 'vitest';
+
+import { valibotContract } from '../contract';
+
+describe('valibot/valibotContract short', () => {
+  test('interprets invalid response as error', () => {
+    const contract = valibotContract(v.string());
+
+    expect(contract.getErrorMessages(2)).toMatchInlineSnapshot(`
+      [
+        "Invalid type",
```

Comment:
🤓 is it possible to retrieve a message with further details from valibot?

Reply:
I'm not sure, cause we mostly use type inference and validation itself (without messages). Writing tests (basically copied them from `zod`) I also asked the same question and after the first glance to Valibot's source code it seems that no, you cannot (you only can provide your own message).
During awaiting for the approval, I'm gonna check it closely

Reply:
Okay, [it's not possible with Valibot out-of-the-box](https://github.com/fabian-hiller/valibot/issues/132) just using `message` property.
Although `Issue` object contains some metadata we can use in order to form our error messages, I don't think it should be included in the package. Some projects might need specific format for error messages.
What we can do I think is to pass an optional callback (or an `options` object with kind of `errorMessage` property) to the `valibotContract` function, that receives an `Issue` object and return an error message. It will be flexible and non-opinionated.

What do you think?

Reply:
Sorry for the late replay.

I do not think it's a good idea to provide a callback for now. No one was asking for this feature, so why need we to introduce it? Let's stick with some reasonable defaults, as it's done in other integrations for contracts.

Reply:
Sounds good. I also prefer to start with what Valibot provides out-of-the-box

Reply:
However, it would be nice to improve default messages because "invalid type" is not a good result of contract application 🤓

Reply:
Ah, misunderstood you a bit. Gonna improve it soon 👍